core/src/main/java/org/infinispan/interceptors/distribution/DistributionBulkInterceptor.java

```diff
@@ -378,4 +381,20 @@ public TxBackingKeySet(Cache<K, V> cache, CacheSet<CacheEntry<K, V>> entrySet, F
                  StreamMarshalling.entryToKeyFunction(), ctx);
       }
    }
+
+   @Override
+   public Object visitGetKeyValueCommand(InvocationContext ctx, GetKeyValueCommand command)
+         throws Throwable {
+      return invokeNext(ctx, command);
```

Comment:
I'm removing this, it didn't make a significant difference - in fact, not even removing the DistributionBulkInterceptor completely didn't improve the results on my local machine.